docs/data-sources/notification_destinations.md

```diff
@@ -0,0 +1,60 @@
+---
+subcategory: "Workspace"
+---
+# databricks_notification_destination Data Source
+
+-> **NOTE** Ensure that the notification destinations you are attempting to retrieve have been created and are accessible within your workspace.
+
+This data source allows you to retrieve information about [Notification Destinations](https://docs.databricks.com/api/workspace/notificationdestinations). Notification Destinations are used to send notifications for query alerts and jobs to external systems such as email, Slack, Microsoft Teams, PagerDuty, or generic webhooks. 
+
+## Example Usage
+
+
+```hcl
+resource "databricks_notification_destination" "email" {
+  display_name = "Email Destination"
+  config {
+    email {
+      addresses = ["abc@gmail.com"]
+    }
+  }
+}
+
+resource "databricks_notification_destination" "slack" {
+  display_name = "Slack Destination"
+  config {
+    slack {
+      url = "https://hooks.slack.com/services/..."
+    }
+  }
+}
+
+data  "databricks_notification_destinations" "this" {} # Lists all notification desitnations
+
+data "databricks_notification_destinations" "filtered_notification" {
+    display_name_contains = "Destination"
+    type = "EMAIL"
+}
+
+```
+
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `display_name_contains` - (Optional) A substring to filter Notification Destinations by their display name. 
```

Comment:
is it a case-sensitive match or case-insensitive? We need to explicitly mention it

Reply:
Also we need to document what will happen if there is no match - do we throw an error, or just return an empty list?